extensions/load-balancer/gce-extension/src/main/java/org/apache/stratos/gce/extension/GCELoadBalancer.java

```diff
@@ -0,0 +1,391 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.stratos.gce.extension;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.stratos.gce.extension.config.GCEClusterConfigurationHolder;
+import org.apache.stratos.gce.extension.config.GCEContext;
+import org.apache.stratos.gce.extension.util.GCEOperations;
+import org.apache.stratos.load.balancer.common.domain.*;
+import org.apache.stratos.load.balancer.extension.api.LoadBalancer;
+import org.apache.stratos.load.balancer.extension.api.exception.LoadBalancerExtensionException;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.*;
+
+public class GCELoadBalancer implements LoadBalancer {
+
+    private static final Log log = LogFactory.getLog(GCELoadBalancer.class);
+    //PROTOCOL should be TCP or UDP
+    private static final String PROTOCOL = "TCP";
+    private GCEOperations gceOperations;
+    /**
+     * One configuration object per cluster will be created
+     * One cluster  has one target pool,one forwarding rule and a health check
+     * This hash map is used to hold cluster IDs and corresponding configuration
+     */
+    private Map<String, GCEClusterConfigurationHolder> clusterToLoadBalancerConfigurationMap;
+
+    public GCELoadBalancer() throws IOException, GeneralSecurityException {
+        gceOperations = new GCEOperations();
+        clusterToLoadBalancerConfigurationMap = new HashMap<String, GCEClusterConfigurationHolder>();
+
+    }
+
+    /**
+     * Listen to latest topology and update load balancer configuration
+     *
+     * @param topology latest topology to be configured
+     * @return - true - if the load balancer was successfully configured. else false
+     * @throws LoadBalancerExtensionException
+     */
+    @Override
+    public boolean configure(Topology topology) throws LoadBalancerExtensionException {
+        log.info("Complete topology received. Configuring Load balancer ");
```

Comment:
If the info log says configuring, may be we need to add three dots at the end of the sentence and remove the blank character.